GeneratorInterface/RivetInterface/plugins/GenParticles2HepMCConverter.cc

```diff
@@ -81,15 +89,19 @@ void GenParticles2HepMCConverter::produce(edm::Event& event, const edm::EventSet
 
   auto const& pTableData = eventSetup.getData(pTable_);
 
-  HepMC::GenEvent* hepmc_event = new HepMC::GenEvent();
+  HepMC3::GenEvent* hepmc_event = new HepMC3::GenEvent();
   hepmc_event->set_event_number(event.id().event());
-  hepmc_event->set_signal_process_id(genEventInfoHandle->signalProcessID());
-  hepmc_event->set_event_scale(genEventInfoHandle->qScale());
-  hepmc_event->set_alphaQED(genEventInfoHandle->alphaQED());
-  hepmc_event->set_alphaQCD(genEventInfoHandle->alphaQCD());
+  hepmc_event->add_attribute("signal_process_id",
+                             std::make_shared<HepMC3::IntAttribute>(genEventInfoHandle->signalProcessID()));
+  hepmc_event->add_attribute("event_scale", std::make_shared<HepMC3::DoubleAttribute>(genEventInfoHandle->qScale()));
+  hepmc_event->add_attribute("alphaQCD", std::make_shared<HepMC3::DoubleAttribute>(genEventInfoHandle->alphaQCD()));
+  hepmc_event->add_attribute("alphaQED", std::make_shared<HepMC3::DoubleAttribute>(genEventInfoHandle->alphaQED()));
 
   hepmc_event->weights() = genEventInfoHandle->weights();
 
+  // resize cross section to number of weights
+  xsec_->set_cross_section(std::vector<double>(hepmc_event->weights().size(), xsec_->xsec(0)),
+                           std::vector<double>(hepmc_event->weights().size(), xsec_->xsec_err(0)));
```

Comment:
@mseidel42 , looks like IB relvals are failing due to ` xsec_->xsec_err(0)` [a] and `cross_section_errors.size()` is 0.


[a]
https://gitlab.cern.ch/hepmc/HepMC3/-/blob/master/include/HepMC3/GenCrossSection.h?ref_type=heads#L176-180

Reply:
Hi Shahzad, yes, it's a problem with particle guns because they do not provide valid event weights.

Working on a fix, will submit it soon!

Reply:
https://github.com/cms-sw/cmssw/pull/45367